src/components/sortControl/sortControl.test.tsx

```diff
@@ -0,0 +1,51 @@
+import { describe, expect, test } from 'vitest';
+import {
+  screen,
+  render,
+  fireEvent,
+  getByTestId,
+  waitFor,
+} from '@testing-library/react';
+import SortControl from './SortControl';
+import user from '@testing-library/user-event';
+import { vi } from 'vitest';
+
+let sortOptions: movieSortOption[];
+describe('Genre Select tests', () => {
+  beforeEach(() => {
+    sortOptions = ['Release Date', 'Title'];
+  });
+
+  test('check all sort options are added correctly', () => {
+    const SelectSortOptionHandler = vi.fn();
+    render(
+      <SortControl
+        sortOptions={sortOptions}
+        defaultSelection={sortOptions[0]}
+        onSelect={SelectSortOptionHandler}
+      />
+    );
+    const sortOptionsElements = screen.getAllByTestId((value) =>
+      value.startsWith('sortOption:')
+    );
+    expect(sortOptionsElements.length).toBe(2);
```

Comment:
The test doesn't check the desired goal properly. It checks the count of options rendered but it needs to check the value as well.